src/types/any.rs

```diff
@@ -488,91 +523,104 @@ impl PyAny {
     ///
     /// This is equivalent to the Python expression `self.name(*args, **kwargs)`.
     ///
+    /// To avoid repeated temporary allocations of Python strings, the [`intern!`] macro can be used
+    /// to intern `name`.
+    ///
     /// # Examples
     ///
     /// ```rust
     /// use pyo3::prelude::*;
-    /// use pyo3::types::{IntoPyDict, PyList};
+    /// use pyo3::types::PyDict;
     ///
-    /// # fn main() -> PyResult<()> {
-    /// Python::with_gil(|py| -> PyResult<()> {
-    ///     let list = PyList::new(py, vec![3, 6, 5, 4, 7]);
-    ///     let kwargs = vec![("reverse", true)].into_py_dict(py);
+    /// const CODE: &str = r#"
+    /// class A:
+    ///     def method(self, *args, **kwargs):
+    ///         assert args == ("hello",)
+    ///         assert kwargs == {"cruel": "world"}
+    ///         return "called with args and kwargs"
+    /// a = A()
+    /// "#;
     ///
-    ///     list.call_method("sort", (), Some(kwargs))?;
-    ///     assert_eq!(list.extract::<Vec<i32>>()?, vec![7, 6, 5, 4, 3]);
+    /// # fn main() -> PyResult<()> {
+    /// Python::with_gil(|py| {
+    ///     let module = PyModule::from_code(py, CODE, "", "")?;
+    ///     let instance = module.getattr("a")?;
+    ///     let args = ("hello",);
+    ///     let kwargs = PyDict::new(py);
+    ///     kwargs.set_item("cruel", "world")?;
+    ///     let result = instance.call_method("method", args, Some(kwargs))?;
```

Comment:
Maybe this would be a good place to showcase using the `intern!` macro to make it easier for the reader of the documentation to understand how it is supposed to be used?